### ISSUE TITLE
Language server reporting issues not present on commandline

### ISSUE THREAD

**Issue by Swoogan** [microsoft/pyright#4652]
**Describe the bug**
With `typeCheckingMode` set to `off`, unused variables and imports will get reported from the language server, but not when running from the command line.

**To Reproduce**
Set `typeCheckingMode` to `off`, and create a file `test.py` that has an unused import:

```
import os
```
Open file with LSP client and you will see (in neovim): `test.py|1 col 8-10 note| "os" is not accessed`

Run `pyright test.py` and the output will say:

```
Found 1 source file
pyright 1.1.258
0 errors, 0 warnings, 0 informations
```

Setting `reportUnusedImport` to `none` has no effect, which is to be expected as it should be off with type checking disabled.

**Expected behavior**
I would expect that with type checking disabled, only syntax errors and warnings would be displayed, or that there would be some way to disable this behaviour.

**VS Code extension or command-line**
Running from neovim v0.8.2 with pyright 1.1.258

**Additional context**
I would like to disable these notes as they are provided to me by another linter, so I end up with duplicate diagnostics.

**Comment by erictraut** [comment]
This is as designed. The `"os" is not accessed` message is part of a "hint", not a diagnostic. It is provided only if the language server client indicates that it wants such hints. Clients should not treat such hints as diagnostics. They typically display these hints as "grayed-out" text. If your client is requesting these hints and then displaying them as diagnostics, I consider that a bug, and you should report it to the maintainers of your client or LSP plugin.

**Comment by Swoogan** [comment]
Hi. Thanks for the prompt reply and the hint about hints. However, I am confused by these comments:

> If your client is requesting these hints and then displaying them as diagnostics, I consider that a bug, and you should report it to the maintainers of your client or LSP plugin.
> 
> If your client isn't going to handle these hints correctly, it can simply not request them from the language server.

The LSP spec shows the type [Diagnostics](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#diagnostic) includes a [DiagnosticSeverity](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#diagnosticSeverity) with `Hint` in the type. Also, it seems the server sends these diagnostics to the client, rather than the client requesting them, as per [PublishDiagnostics](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_publishDiagnostics). So I don't think the client requests them, and I don't think handling the hints as diagnostics seems contrary to the spec. Am I misunderstanding it?

I wonder if there is some way to tell the server that the client can't handle diagnostics to prevent it from publishing them? Otherwise, I will configure my client to intercept the "PublishDiagnostics Notification" and do nothing in the case that the client is connected to `pyright`.

**Comment by erictraut** [comment]
The hints in this case have a special "tag" that indicates that the client should interpret them not as normal diagnostics. Clients can specify which "tags" they support, and pyright will honor this list. If the client indicates that it doesn't support the tag, pyright won't generate these hints. You client (or LSP plugin) is requesting hints with these tags.

**Comment by Swoogan** [comment]
So it turns out that it is very easy to disable these tags within the `nvim-lspconfig` plugin:
```
local capabilities = vim.lsp.protocol.make_client_capabilities()
capabilities.textDocument.publishDiagnostics.tagSupport.valueSet = { 2 } 
require('lspconfig')['pyright'].setup{
    capabilities = capabilities,
    on_attach = on_attach
}
```
If you want to disable all type checking and diagnostics, you can either use:
```
require('lspconfig')['pyright'].setup {
  on_attach = function(client, buffer)
    client.handlers["textDocument/publishDiagnostics"] = function(...) end
    on_attach(client, buffer)
  end,
  settings = {
    python = {
      analysis = {
        typeCheckingMode = "off",
      }
    }
  }
}
```
Or turn off typeChecking, clear the tagSupport, and turn off all the syntax checking individually (eg: `reportUndefinedVariable`, `reportMissingImports`).

**Comment by NeilGirdhar** [comment]
@swoogan Would you by any chance know how to do it with the moder overriding interface:
```lua
        local pyright_capabilities = require('cmp_nvim_lsp').default_capabilities{
           tagSupport = { valueSet = { 2 } }
        }
```
doesn't seem to work for me

**Comment by Swoogan** [comment]
@NeilGirdhar I have something setup along the lines of 

```
        local cap = require('cmp_nvim_lsp').default_capabilities()
        cap.textDocument.publishDiagnostics = { tagSupport = { valueSet = { 2 } } }
```
